src/pages/ProductDetail/index.tsx

```diff
@@ -9,41 +9,58 @@ import { Product, Loading, ProductList } from '@/components';
 const ProductDetail = () => {
   const { id } = useParams<{ id: string }>();
 
+  // Use the useProductById hook for the product and status
+  const { product, isLoading: isProductLoading, status } = useProductById(id || '');
+
   // Use the useProducts hook for related products
-  const { product, isLoading: isProductLoading } = useProductById(id || '');
   const { products, isLoading: isRelatedProductsLoading, hasMore, onLoadMore } = useProducts(product?.category || '');
 
+  // Determine loading state
+  const isLoading = isProductLoading && status === undefined;
```

Comment:
Do we need to check status for flag isLoading?

Reply:
Fixed